src/storage/txn/latch.rs

```diff
@@ -139,12 +139,13 @@ impl Latches {
     }
 
     /// Creates a lock which specifies all the required latches for a command.
-    pub fn gen_lock<H>(&self, keys: &[H]) -> Lock
+    pub fn gen_lock<'a, H: 'a, I>(&'a self, keys: I) -> Lock
     where
         H: Hash,
+        I: Iterator<Item = &'a H>,
```

Comment:
You can use `IntoIterator` so that changes to https://github.com/tikv/tikv/pull/8167/files#diff-21f8c57a4ef3ae7c3bb60229864218d0R345 and https://github.com/tikv/tikv/pull/8167/files#diff-21f8c57a4ef3ae7c3bb60229864218d0R350 are unnecessary.